app/uk/gov/hmrc/fileupload/virusscan/ScannerActor.scala

```diff
@@ -68,6 +71,7 @@ class ScannerActor(subscribe: (ActorRef, Class[_]) => Boolean,
           scanNext()
         case Xor.Left(ScanResultVirusDetected) =>
           notify(hasVirus = true)
+          deleteInfectedFile()
```

Comment:
I wonder if we should directly and synchronously delete the file in this actor, or we should have separate actor that reacts to "MarkFileAsInfected" event.
(as we do for clean files - in TransferAgent")

The only difference I see is that with actor approach we don't have to wait until the file is deleted.